src/evict/evict_lru.c

```diff
@@ -1785,6 +1785,17 @@ __evict_walk(WT_SESSION_IMPL *session, WTI_EVICT_QUEUE *queue)
             }
         }
 
+        /*
+         * If the cache walk flags have changed since the prior eviction pass on this tree then
+         * reset the walk effectiveness tracking. Imagine a case where only dirty content has been
+         * looked for and this tree doesn't have much dirty content. Then eviction starts looking
+         * for clean content - this tree might be a cornucopia of good clean candidate pages.
+         */
+        if (btree->last_evict_walk_flags != evict->flags) {
```

Comment:
Can we be more specific here? The last time the old evict flag may be dirty hard and this time it is dirty. I don't think we should reset the walk period in this case.

Reply:
We might actually want to reset in that case. We could have been including additional pages when the HARD threshold was met, but will start skipping them when HARD is cleared. Starting from a non-biased walk when the type of content being searched for is different felt reasonable to me.
OTOH: It could be a bit surprising - for example, enabling/disabling scrub is more of an advisory flag re: how to do eviction, rather than an indication of what types of content we should be looking for. I could be convinced this should be restricted to the CLEAN/DIRTY/UPDATES flags for that reason.